modules/api/pkg/ee/clusterbackup/storage-location/provider.go

```diff
@@ -110,57 +141,78 @@ func (p *BackupStorageProvider) CreateUnsecured(ctx context.Context, cbslName, p
 		},
 		Key: "cloud-credentials",
 	}
-	if err := p.privilegedClient.Create(ctx, cbsl); err != nil {
+	if err := client.Create(ctx, cbsl); err != nil {
+		return nil, common.KubernetesErrorToHTTPError(err)
+	}
+	ownerReference := metav1.OwnerReference{
+		APIVersion: ownerReferenceApi,
+		Kind:       ownerReferenceKind,
+		Name:       cbslFullName,
+		UID:        cbsl.UID,
+	}
+
+	secret.ObjectMeta.OwnerReferences = append(secret.ObjectMeta.OwnerReferences, ownerReference)
+
+	if err := p.privilegedClient.Create(ctx, secret); err != nil {
 		return nil, err
 	}
 	return cbsl, nil
 }
 
-func (p *BackupStorageProvider) DeleteUnsecured(ctx context.Context, name string) error {
+func (p *BackupStorageProvider) Delete(ctx context.Context, userInfo *provider.UserInfo, name string) error {
+	if userInfo == nil {
+		return errors.New("a user is missing but required")
+	}
+
+	client := p.privilegedClient
+
 	cbsl := &kubermaticv1.ClusterBackupStorageLocation{}
-	if err := p.privilegedClient.Get(ctx, types.NamespacedName{Name: name, Namespace: resources.KubermaticNamespace}, cbsl); err != nil {
+	if err := client.Get(ctx, types.NamespacedName{Name: name, Namespace: resources.KubermaticNamespace}, cbsl); err != nil {
 		if apierrors.IsNotFound(err) {
 			return nil
 		}
 		return err
 	}
```

Comment:
Is there a reason why the priviledged client is needed for getting the ClusterBackupStorageLocation? Couldn't the impersonated client be used for this?

Reply:
refactored to apply use the same client for both operations based on the user role.